Toggle Feature Flag in Launchdarkly (#16)
port-labs/self-service-actions-examples: github-workflows/launchdarkly/README.md

This GitHub action allows you to quickly toggle LaunchDarkly Feature Flags via Port Actions with ease.

## Inputs
| Name                 | Description                                                                                          | Required | Default            |
|----------------------|------------------------------------------------------------------------------------------------------|----------|--------------------|
| feature_flag_key         | LaunchDarkly Feature Flag Key                                                       | true    | -                  |
| projectKey              | The LaunchDarkly Project Key where the feature flag exists                                                     | true     | default                  |
| environment_key              | The LaunchDarkly Environment Key where the flag exists                                                              | true    | test               |
| flag_state           | Desired state of the feature flag (true for enabled, false for disabled)                                                | true     | true                  |

> ## Quickstart - Toggle a lauchdarkly feature flag from the service catalog

Reviewer: ```suggestion
## Toggle a LauchDarkly feature flag from the service catalog

```

Follow these steps to get started with the Golang template

1. Create the following GitHub action secrets
* `LAUNCHDARKLY_ACCESS_TOKEN` - a token with permission to toggle a feature flag in LaunchDarkly [learn more](https://docs.launchdarkly.com/home/account-security/api-access-tokens)
* `PORT_CLIENT_ID` - Port Client ID [learn more](https://docs.getport.io/build-your-software-catalog/sync-data-to-catalog/api/#get-api-token)
* `PORT_CLIENT_SECRET` - Port Client Secret [learn more](https://docs.getport.io/build-your-software-catalog/sync-data-to-catalog/api/#get-api-token) 

> 2. Install the Ports GitHub app from [here](https://github.com/apps/getport-io/installations/new).

Reviewer: ```suggestion
2. Install the Port GitHub app from [here](https://github.com/apps/getport-io/installations/new).

```

3. Install Port's LaunchDarkly integration [learn more](https://github.com/port-labs/Port-Ocean/tree/main/integrations/launchdarkly)

> >**Note** This step is not required for this example, but it will create all the blueprint boilerplate for you, and also update the feature flag catalog in real time.

Reviewer: ```suggestion
>**Note** The above step is not required for this example, but it will create all the blueprint boilerplate for you, and also update the feature flag catalog in real time.

```

        "description": "LaunchDarkly Project Key",
        "title": "project_key",
        "icon": "Launchdarkly",
        "type": "string",
        "default": "default"
      },
      "environment_key": {
        "description": "LaunchDarkly Environment Key where the flag exists",
        "title": "environment_key",
        "icon": "Launchdarkly",
        "type": "string",
        "default": "test"
      },
      "flag_state": {
        "title": "flag_state",
        "description": "Desired state of the feature flag (true for enabled, false for disabled)",
        "icon": "Launchdarkly",
        "type": "boolean",
        "default": true
      }
    },
    "required": [
      "feature_flag_key",
      "project_key"
    ],
    "order": [
      "feature_flag_key",
      "project_key",
      "environment_key"
    ]
  },
  "invocationMethod": {
    "type": "GITHUB",
    "org": "your-github-org",
    "repo": "your-github-repository",
    "workflow": "toggle-feature-flag.yaml",
    "omitUserInputs": false,
    "omitPayload": false,
    "reportWorkflowStatus": true
  },
  "trigger": "DAY-2",
  "description": "Toggle a Feature Flag in launchdarkly",
  "requiredApproval": false
}
```

> >**Note** Replace the invocation method with your own repository details.

Reviewer: ```suggestion
>**Note** In the `invocationMethod` section, replace the `org` and `repo` values with your own repository details

```

5. Create a workflow file under .github/workflows/toggle-feature-flag.yaml with the following content:

> ```yml

Reviewer: Do not paste the file content here, add a link to the file that is saved in the repository. Having the file twice will make maintenance difficult when changes are made
